src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -255,6 +296,21 @@ public FileCharacteristics GetFileCharacteristics(string fileName)
         }
 
         /// <summary>
+        /// Invalidates appenders for all files that were archived.
+        /// </summary>
+        public void InvalidateAppendersForInvalidFiles()
```

Comment:
I doubting if we should expose a public method when I has no body. So move `#if !SILVERLIGHT && !__IOS__ && !__ANDROID__` outside this method. What do you think?